### PR TITLE
Feature/data counter

### PR DESCRIPTION
**Pull request by juanmaguitar**
- ⚙️Task → https://jira.scmspain.com/browse/SUIC-50
- 📄Paper → https://paper.dropbox.com/doc/SUI-Data-counter-TqR9qBw4WRr5l5gNMvvEE
- 🌍Online Demo → https://sui-components-feature-data-counter.now.sh/workbench/molecule/dataCounter/demo

### CODE REVIEW

File: components/molecule/dataCounter/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "main": "lib/index.js",
+  "scripts": {
+    "build": "npx rimraf ./lib && npx mkdirp ./lib && npm run build:js && npm run build:styles",
+    "build:js": "../../../node_modules/.bin/babel --presets sui ./src --out-dir ./lib",
+    "build:styles": "../../../node_modules/.bin/cpx './src/**/*.scss' ./lib"
+  },
+  "dependencies": {
+    "@s-ui/component-dependencies": "1",
+    "@s-ui/hoc": "^1.18.0",
+    "@s-ui/react-atom-input": "^3.9.0",
```
Comment (jcger) [inline components/molecule/dataCounter/package.json]: please remove the caret and use major version for our packages

File: components/molecule/dataCounter/src/index.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        <div
+          className={cx(
+            CLASS_INPUT_CONTAINER,
+            `${CLASS_INPUT_CONTAINER}--${size}`
+          )}
+        >
+          <AtomButton
+            disabled={disabled || value === min}
+            onClick={decrementValue}
+            type="secondary"
```
Comment (jcger) [inline components/molecule/dataCounter/src/index.js]: you should use a constant for both button type props are it should be always the same

File: components/molecule/dataCounter/src/index.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+          >
+            -
+          </AtomButton>
+          <AtomInput
+            id={id}
+            disabled={disabled}
+            size={size}
+            charsSize={charsSize}
+            placeholder={placeholder}
+            value={value}
```
Comment (jcger) [inline components/molecule/dataCounter/src/index.js]: input propType is string or bool but this is a number https://github.com/SUI-Components/sui-components/blob/master/components/atom/input/src/index.js#L88

File: components/molecule/dataCounter/src/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+  const incrementValue = e => {
+    if (isBelowMaxValue) onChange(e, {value: value + 1})
+  }
+
+  const decrementValue = e => {
+    if (isOverMinValue) onChange(e, {value: value - 1})
+  }
+
+  const handleKeyDown = e => {
```
Comment (rmoralp) [inline components/molecule/dataCounter/src/index.js]: Copied from Paper: "Input field value can also be changed by the user using the keyboard so Input field must receive focus as well as the buttons, but only number values are allowed."

Should numbers keys have to be allowed? I think that should be

Comment (juanmaguitar) [reply components/molecule/dataCounter/src/index.js]: Done!

File: components/molecule/dataCounter/src/index.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            -
+          </AtomButton>
+          <AtomInput
+            id={id}
+            disabled={disabled}
+            size={size}
+            charsSize={charsSize}
+            value={internalValue}
+            onKeyDown={handleKeyDown}
+            onChange={handleKeyDown}
```
Comment (rmoralp) [inline components/molecule/dataCounter/src/index.js]: onChange I think that is doesn't needed here. It already has onKeyDown

File: components/molecule/dataCounter/src/index.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+  const addDigit = (e, {key: digit}) => {
+    const newValue = internalValue + digit
+    if (newValue.length <= 2) {
+      setInternalValue(newValue)
+      onChange(e, {value: newValue})
+    }
+  }
+
+  const handleKeyDown = e => {
```
Comment (rmoralp) [inline components/molecule/dataCounter/src/index.js]: Why not using {value} coming from AtomInput?
I can't add a digit before other. It always is being added at the end

**Review by jcger — DISMISSED** [review]

File: components/molecule/dataCounter/src/index.js (hunk 6 above)
Comment (juanmaguitar) [reply components/molecule/dataCounter/src/index.js]: Right! I have improved both the code and the UX based on your comments

File: components/molecule/dataCounter/src/index.js [hunk 7]
```diff
@@ -67,28 +65,19 @@ const MoleculeDataCounter = ({
...
-
-  const addDigit = (e, {key: digit}) => {
-    const newValue = internalValue + digit
-    if (newValue.length <= 2) {
-      setInternalValue(newValue)
-      onChange(e, {value: newValue})
+  const handleChange = (e, {value}) => {
+    const numericValue = parseInt(value, 10)
+    if (value.length <= 2 && !isNaN(numericValue)) {
+      const strNewValue = String(numericValue)
```
Comment (rmoralp) [inline components/molecule/dataCounter/src/index.js]: This 3 lines are repeated 3 times in`incrementValue`, `decrementValue` and here `handleChange`. What do you think about wrap it in a function and reutilize this code?

Comment (juanmaguitar) [reply components/molecule/dataCounter/src/index.js]: done!

**Review by rmoralp — DISMISSED** [review]

File: components/molecule/dataCounter/src/index.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+  const incrementValue = e => {
+    if (isBelowMaxValue) {
+      const nValue = internalValue === '' ? min : parseInt(internalValue) + 1
+      assignValue(e, {nValue})
+    }
+  }
+
+  const decrementValue = e => {
+    if (isOverMinValue) {
+      const nValue = internalValue === '' ? min : parseInt(internalValue) - 1
```
Comment (jcger) [inline components/molecule/dataCounter/src/index.js:65]: you are repeating this code, you should move it into its own function

File: components/molecule/dataCounter/src/index.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+            value={internalValue}
+            onKeyDown={handleKeyDown}
+            onChange={handleChange}
+          />
+          <AtomButton
+            disabled={incrementDisabled}
+            onClick={incrementValue}
+            type="secondary"
+          >
+            +
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/dataCounter/src/index.js:130]: I don't know. Should they?

**Review by jcger — APPROVED** [review]

**Review by rmoralp — APPROVED** [review]

**Pull request merged**